p2p/pex/errors.go

```diff
@@ -72,3 +72,12 @@ type ErrAddressBanned struct {
 func (err ErrAddressBanned) Error() string {
 	return fmt.Sprintf("Address: %v is currently banned", err.Addr)
 }
+
+// ErrUnsolicitedList is thrown when a peer provides a list of addresses that have not been asked for.
+type ErrUnsolicitedList struct {
```

Comment:
again, I don't understand a need for exposing this error

why not

```go
var errUnsolicitedList = errors.New("unsolicited pexAddrsMessage")
```

Reply:
I'm easy either way, but bez was saying in the other PR that it's good to expose typed errors so that external logic can use them to check the errors. With the check evidence PR, I changed an error back to public again so the RPC could use them. Perhaps it would be a good idea to make a clear ruling of when to use private or public error types

Reply:
Okay, so from what I understand, `ErrUnsolicitedList` is a potential error from the public function `RecieveAddrs` and so should therefore be public

Reply:
I could drop the peer as a variable of the error as this is part of the input of the function and so we can assume the external logic already has it.

Reply:
ErrUnsolicitedList{src} - caller should add src if needed